base/pkg/entry.jl

```diff
@@ -190,6 +190,9 @@ end
 
 function clone(url::AbstractString, pkg::AbstractString)
     info("Cloning $pkg from $url")
+    if !(':' in url) && !isfile(joinpath("METADATA",pkg,"url"))
+        throw(PkgError("package $pkg at url $url doesn't appear to exist"))
```

Comment:
This isn't technically always a `url`, it could be a local path. And I think with the error done here, it doesn't need to be explicitly checked for in the one-argument `clone` any more?